packages/contracts-bridge/contracts/test/TokenRegistry.t.sol

```diff
@@ -1,10 +1,558 @@
+        address repr = tokenRegistry.getRepresentationAddress(
+            remoteDomain,
+            remoteTokenRemoteAddress
+        );
+        assertEq(repr, remoteTokenLocalAddress);
+    }
+
+    function test_getRepresentationAddressWrongDetailsReturnZeroFuzzed(
+        uint32 domain,
+        bytes32 randomAddress
+    ) public {
+        vm.assume(
+            randomAddress != remoteTokenRemoteAddress || domain != remoteDomain
+        );
+        address repr = tokenRegistry.getRepresentationAddress(
+            domain,
+            randomAddress
+        );
+        assertEq(repr, address(0));
+    }
+
+    function test_getRepresentationAddressFuzzed(
+        uint32 fuzzedDomain,
+        bytes32 fuzzedId
+    ) public {
+        address localAddress = createRemoteToken(fuzzedDomain, fuzzedId);
+        address repr = tokenRegistry.getRepresentationAddress(
+            fuzzedDomain,
+            fuzzedId
+        );
+        assertEq(repr, localAddress);
+    }
+
+    event TokenDeployed(
+        uint32 indexed domain,
+        bytes32 indexed id,
+        address indexed representation
+    );
+
+    function test_ensureLocalTokenDeploy() public {
+        uint32 newDomain = 13;
+        bytes32 newId = bytes32("hey yoou");
+        // It's the second contract that is been deployed by tokenRegistry
+        // It deploys a bridgeToken during setUp() of BridgeTest
+        address calculated = computeCreateAddress(address(tokenRegistry), 2);
+        vm.expectEmit(true, true, true, false);
+        emit TokenDeployed(newDomain, newId, calculated);
+        vm.prank(tokenRegistry.owner());
+        address deployed = tokenRegistry.ensureLocalToken(newDomain, newId);
+        assertEq(deployed, calculated);
+    }
+
+    function test_ensureLocalTokenExisting() public {
+        vm.prank(tokenRegistry.owner());
+        address addr = tokenRegistry.ensureLocalToken(
+            remoteDomain,
+            remoteTokenRemoteAddress
+        );
+        assertEq(addr, remoteTokenLocalAddress);
+    }
+
+    function test_ensureLocalTokenOnlyOwner() public {
+        // It's the second contract that is been deployed by tokenRegistry
+        // It deploys a bridgeToken during setUp() of BridgeTest
+        vm.expectRevert("Ownable: caller is not the owner");
+        tokenRegistry.ensureLocalToken(remoteDomain, remoteTokenRemoteAddress);
+    }
+
+    function test_ensureLocalTokenDeployFuzzed(uint32 domain, bytes32 id)
+        public
+    {
+        vm.startPrank(tokenRegistry.owner());
+        if (domain == localDomain) {
+            assertEq(
+                tokenRegistry.ensureLocalToken(domain, id),
+                TypeCasts.bytes32ToAddress(id)
+            );
+            return;
+        }
+        // It's the second contract that is been deployed by tokenRegistry
+        // It deploys a bridgeToken during setUp() of BridgeTest
+        address calculated = computeCreateAddress(address(tokenRegistry), 2);
```

Comment:
us `vm. getNonce` ?

Reply:
it gets the job done, so why should we add that?